src/solvers/flattening/arrays.cpp

```diff
@@ -111,21 +111,20 @@ void arrayst::collect_arrays(const exprt &a)
     if(a.operands().size()!=3)
```

Comment:
This check is now unnecessary given the use of `to_with_expr` (which does that check).